### PR TITLE
acristinasoara code review

### PR DESCRIPTION
**Pull request by PerryBaran**

### CODE REVIEW

File: src/kata2.booleanToWord.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
-const booleanToWord = (boolean) => {};
+const booleanToWord = (boolean) => {
+  return boolean === true ? "Yes" : "No";
```
Comment (PerryBaran) [inline src/kata2.booleanToWord.js:2]: Good use of the ternary operator. However, you don't need to check if a boolean is `===` to `true`, as a boolean value is already either `true` or `false` you can shorten this to:
```
return boolean ? "Yes" : "No"
```

File: test/kata2.booleanToWord.test.js [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
 
 describe("booleanToWord", () => {
-  // how do we create specs again???
+  test("returns Yes when passed True", () => {
+    expect(booleanToWord(true)).toBe("Yes");
+  });
+
+  test("returns No when passed False", () => {
+    expect(booleanToWord(false)).toBe("No");
+  });
```
Comment (PerryBaran) [inline test/kata2.booleanToWord.test.js:10]: Good tests - the only very minor change I would make is to add quotation marks around the words `Yes` and `No` in your test description, to make it even more obvious that this tests for a string e.g.
```
test("returns 'Yes' when passed True", () => {...
```

File: src/kata1.fizzBuzz.js [hunk 3]
```diff
@@ -1,3 +1,18 @@
...
+  }
+  if (isDivisableBy5) {
+    return "Buzz";
+  }
+
+  return number;
+};
 
 module.exports = fizzBuzz;
+//
```
Comment (PerryBaran) [inline src/kata1.fizzBuzz.js:18]: remove empty comment on line 18

File: test/kata4.humanCatDogYears.test.js [hunk 4]
```diff
@@ -1,3 +1,8 @@
 const { humanCatDogYears } = require("../src");
 
-// Look Ma, no handlebars!!!
+describe("humanCatDogYears", () => {
+  it("returns array of human, cat and dog years when passed human years", () => {
+    expect(humanCatDogYears(10)).toEqual([10, 56, 64]);
+    expect(humanCatDogYears(20)).toEqual([20, 96, 114]);
+  });
+});
```
Comment (PerryBaran) [inline test/kata4.humanCatDogYears.test.js:8]: You should add more tests for the edge cases. For example, if human years is `0`, `1` or `2`.

File: src/kata4.humanCatDogYears.js [hunk 5]
```diff
@@ -1,3 +1,7 @@
-const humanCatDogYears = (number) => {};
+const humanCatDogYears = (number) => {
+  const humanCatYears = 15 + 9 + (number - 2) * 4;
+  const humanDogYears = 15 + 9 + (number - 2) * 5;
+  return [number, humanCatYears, humanDogYears];
+};
```
Comment (PerryBaran) [inline src/kata4.humanCatDogYears.js:5]: Although this returns the expected result if `number` is equal to `2` or more, you return the wrong result for `number` equal to `0` or `1`. The easiest way to fix this would be to add two `if` statements at the top of your function e.g.
```
const humanCatDogYears = (number) => {
  if (number === 0) return [0, 0, 0];
  if (number === 1) return [1, 15, 15];
  ...
};
```

File: src/kata5.reachDestination.js [hunk 6]
```diff
@@ -1,3 +1,6 @@
-const reachDestination = (distance, speed) => {};
+const reachDestination = (distance, speed) => {
+  const arrivalTime = Math.round((distance / speed) * 2) / 2;
+  return `I should be there in ${arrivalTime} hours.`;
```
Comment (PerryBaran) [inline src/kata5.reachDestination.js:3]: small nitpick, but if `arrivalTime` equals `1`, then `hours` should be singular, i.e.
```
`... 1 hour.`
```
instead of
```
`... 1 hours `
```

File: test/kata5.reachDestination.test.js [hunk 7]
```diff
@@ -1,5 +1,9 @@
 const { reachDestination } = require("../src");
 
 describe("reachDestination", () => {
-  test("returns string with estimated time of arrival", () => {});
+  test("returns string with estimated time of arrival", () => {
+    expect(reachDestination(44, 10)).toBe("I should be there in 4.5 hours.");
+    expect(reachDestination(30, 7)).toBe("I should be there in 4.5 hours.");
+    expect(reachDestination(20, 10)).toBe("I should be there in 2 hours.");
+  });
```
Comment (PerryBaran) [inline test/kata5.reachDestination.test.js:8]: I would add a second test for the edge case, where `arrivalTime` equals `1`, e.g.
```
test("returns string with estimated time of arrival when time to arrival is 1 hour", () => {
  expect(reachDestination(10, 10)).toBe("I should be there in 1 hour.");
}
```

File: src/kata6.joinNames.js [hunk 8]
```diff
@@ -1,3 +1,15 @@
...
+const joinNames = (namesObj) => {
+  let concatenatedNames = "";
+  for (let i = 0; i < namesObj.length; i++) {
+    if (i === namesObj.length - 1) {
+      concatenatedNames += " & " + namesObj[i].name;
+    } else if ( i === namesObj.length - 2) {
+      concatenatedNames += namesObj[i].name;
+    } else {
+      concatenatedNames += namesObj[i].name + ', ';
+    }
```
Comment (PerryBaran) [inline src/kata6.joinNames.js:10]: you could use template literals instead of string concatenation here, e.g.
```
concatenatedNames += ` & ${namesObj[i].name}`;
```

File: test/kata6.joinNames.test.js [hunk 9]
```diff
@@ -1,5 +1,21 @@
...
+      name: "Lisa",
+    };
+
+    const name3 = {
+      name: "Maggie",
+    };
+
+    expect(joinNames([name1, name2, name3])).toBe("Bart, Lisa & Maggie");
+    expect(joinNames([name1, name2])).toBe("Bart & Lisa");
+    expect(joinNames([name2, name3])).toBe("Lisa & Maggie");
```
Comment (PerryBaran) [inline test/kata6.joinNames.test.js:19]: I would separate out these tests into there own assertion e.g.
```
test("returns strings of names, separated by an ampersand when array length is 2", ...);

test("returns strings of names, separated by commas and an ampersand when array length is 3 or more", ...);
```

Comment (PerryBaran) [reply test/kata6.joinNames.test.js:19]: You should also add a test for if the array length is 1 e.g.
```
test("returns name as a string if array length is 1", () => {
  const array = [{ name: "Bart" }];

  expect(joinNames(array)).toBe("Bart");
}
```

File: src/kata6.joinNames.js [hunk 10]
```diff
@@ -1,3 +1,15 @@
...
+  }
+  return concatenatedNames;
```
Comment (PerryBaran) [inline src/kata6.joinNames.js:12]: This works if the `namesObj` has a length greater than `1`, but if it has a length of `1` it currently returns the wrong result. The easiest way to fix this would be to an an `if` statement at the top of your function, e.g
```
const joinNames = (namesObj) => {
  if (namesObj.length === 1) return namesObj[0].name;
  ...
};
```

A better way to solve this problem would be to edit when your adding the `&`. If you change your first two `if` statements to:
```
...
if (i === namesObj.length - 1) {
    concatenatedNames += namesObj[i].name;
} else if ( i === namesObj.length - 2) {
    concatenatedNames += namesObj[i].name + ' & ';
} else {
...
```
i.e. adding the `&` when `i === namesObj.length - 2` instead of when `i === namesObj.length - 1`, then the `&` is only added when the array length is `2` or more.

File: src/kata6.joinNames.js [hunk 11]
```diff
@@ -1,3 +1,15 @@
...
+}
```
Comment (PerryBaran) [inline src/kata6.joinNames.js:13]: You could also use the `reduce` array method for this challenge, e.g
```
const joinNames = (namesObj) => {
  return namesObj.reduce((accumulator, current, index, array) => {
    if (index === array.length - 1) {
      return `${accumulator}${current.name}`;
    }

    if (index === array.length - 2) {
      return `${accumulator}${current.name} & `;
    }

    return `${accumulator}${current.name}, `;
  }, "");
}
```

**Review by PerryBaran — COMMENTED** [review]
Pretty good overall, I've left suggestions throughout on how you can improve it. Some key points + stuff I haven't mentioned elsewhere are:
- You need to add more tests for edge cases. I've left suggestions for tests you could add + ways to change your functions to pass these tests throughout.
- There are some linting (formatting) errors. If you have a red squiggly line or your file name shows up as red in the `EXPLORER`, then you need to fix these issues. If you hover your mouse over the red squiggly line a popup will appear that will give you more information about what is causing the problem. Often there is a button at the bottom of the popup with the text `Quick Fix... (Ctrl+.)` which, if clicked, will usually automatically solve these issues for you.
